plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/names/ClassScopeNameMatcher.kt

```diff
@@ -21,7 +21,7 @@ internal sealed class NameMatcher(val name: MangledName) {
     }
 
     protected val scopedName = name as? ScopedKotlinName
-    protected val packageName = (scopedName?.scope as? PackagePrefixScope)?.packageName
+    protected val packageName = scopedName?.scope?.packageNameIfAny
```

Comment:
Actually it feels like with this pull request we can return to the single `NameMatcher`...